browser-test/echo_client.js

```diff
@@ -0,0 +1,646 @@
+   *
+   * const showcase = require('showcase.v1beta1');
+   *
+   * const client = new showcase.v1beta1.EchoClient({
+   *   // optional auth parameters.
+   * });
+   *
+   * // Iterate over all elements.
+   * client.pagedExpand({})
+   *   .then(responses => {
+   *     const resources = responses[0];
+   *     for (const resource of resources) {
+   *       // doThingsWith(resource)
+   *     }
+   *   })
+   *   .catch(err => {
+   *     console.error(err);
+   *   });
+   *
+   * // Or obtain the paged response.
+   *
+   * const options = {autoPaginate: false};
+   * const callback = responses => {
+   *   // The actual resources in a response.
+   *   const resources = responses[0];
+   *   // The next request if the response shows that there are more responses.
+   *   const nextRequest = responses[1];
+   *   // The actual response object, if necessary.
+   *   // const rawResponse = responses[2];
+   *   for (const resource of resources) {
+   *     // doThingsWith(resource);
+   *   }
+   *   if (nextRequest) {
+   *     // Fetch the next page.
+   *     return client.pagedExpand(nextRequest, options).then(callback);
+   *   }
+   * }
+   * client.pagedExpand({}, options)
+   *   .then(callback)
+   *   .catch(err => {
+   *     console.error(err);
+   *   });
+   */
+  pagedExpand(request, options, callback) {
+    if (options instanceof Function && callback === undefined) {
+      callback = options;
+      options = {};
+    }
+    options = options || {};
+
+    return this._innerApiCalls.pagedExpand(request, options, callback);
+  }
+
+  /**
+   * Equivalent to {@link pagedExpand}, but returns a NodeJS Stream object.
+   *
+   * This fetches the paged responses for {@link pagedExpand} continuously
+   * and invokes the callback registered for 'data' event for each element in the
+   * responses.
+   *
+   * The returned object has 'end' method when no more elements are required.
+   *
+   * autoPaginate option will be ignored.
+   *
+   * @see {@link https://nodejs.org/api/stream.html}
+   *
+   * @param {Object} request
+   *   The request object that will be sent.
+   * @param {string} [request.content]
+   *   The string to expand.
+   * @param {number} [request.pageSize]
+   *   The maximum number of resources contained in the underlying API
+   *   response. If page streaming is performed per-resource, this
+   *   parameter does not affect the return value. If page streaming is
+   *   performed per-page, this determines the maximum number of
+   *   resources in a page.
+   * @param {Object} [options]
+   *   Optional parameters. You can override the default settings for this call, e.g, timeout,
+   *   retries, paginations, etc. See [gax.CallOptions]{@link https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the details.
+   * @returns {Stream}
+   *   An object stream which emits an object representing [EchoResponse]{@link google.showcase.v1beta1.EchoResponse} on 'data' event.
+   *
+   * @example
+   *
```

Comment:
is this identical to the v1beta echo client, wondering if we could instead import the client from here so that they stay in sync?

Reply:
There are some slight changes, mostly in the way we call browser-specific `GrpcClient` functions (lines 100-101, 157-160) and in how `WaitRequest` and `WaitResponse` are created (which I think requires dynamically loading protos in the original echo client)

Reply:
Yeah they are slightly different at this time. We aim to have 100% compatible client working both in browser and in node (will happen after https://github.com/googleapis/gapic-generator/pull/2888 is merged and makes it way through `artman`). After that is done, I'll merge these two clients into one. 

@ramya-ramalingam let's just add a `TODO` comment somewhere here that the two echo client must be merged together.

Reply:
works for me; my concern is that, as they're mostly the same, we will gradually see them diverge over time an aspirational TODO acknowledges this.